scrap_flipkart.py

```diff
@@ -4,37 +4,52 @@
 
 
 def first_method(product_details):
-
-    for product_detail in product_details:
-
-        data = product_detail.findAll("div", {"class":"col col-7-12"})
-
-        for datum in data:
-            name = datum.div
-            print(name.text)  # PRINTING NAME OF EACH PRODUCT FOUND
-
-            rating = datum.find("div", {"class":"hGSR34"})
-            print(rating.text)  # PRINTING RATING OF EACH PRODUCT FOUND
-
-            detail = datum.findAll("li", {"class":"tVe95H"})
-            for k in detail:
+    for i in productDetail:
+        data=i.findAll("div", {"class":"col col-7-12"})
+        for j in data:
+            name=j.div
+            print(name.text)                                #PRINTING NAME OF EACH PRODUCT FOUND 
+            
+            try:
+                rating=j.find("div", {"class":"hGSR34"})
+                print(rating.text)                          #PRINTING RATING OF EACH PRODUCT FOUND
+            except:
+                print("No reviews")
+
+            details=j.findAll("li", {"class":"tVe95H"})     #PRINTING DETAILS OF EACH PRODUCT FOUND
+            for k in details:
                 print(k.text)
-
+                
             print("\n")
 
 
 def second_method(content):
-    product_line = content.findAll("div", {"class":"_3O0U0u"})
-
-    print(len(product_line))
-
-    # This shortcut save I/O Read as I/O is done all at once.
-    product = [j.text for i in product_line for j in i]
-    print(product)
+    productLine=page.findAll("div", {"class":"_3O0U0u"})
+    print(len(productLine))
+    for i in productLine:
+        for j in i:
+            productName=j.div.find("a", {"class":"_2cLu-l"})['title']
+            print (productName)         #PRINTING NAME OF EACH PRODUCT FOUND
+            
+            try:
+                rating=j.div.find("div", {"class":"niH0FQ _36Fcw_"}).text
+                print(rating)           #PRINTING RATING OF EACH PRODUCT FOUND
+            except:
+                print("No reviews")
+
+            try:

```

Comment:
Instead of 2 try blocks try to use a single try block with different exception handling blocks. Try to shift the logic in that case if only a print alert is required in case of the runtime exception.

Reply:
Accepted, I'll make changes for this